File: day-08/part-1/badouralix.go
package main

import (
	"fmt"
	"io/ioutil"
	"os"
	"strconv"
	"strings"
	"time"
)

func run(s string) interface{} {
	split := strings.Split(s, " ")

	tree := make([]int, len(split))
	for i := 0; i < len(split); i++ {
		tree[i], _ = strconv.Atoi(split[i])
	}

	_, result := rec(0, 0, tree)
	return result
}

func rec(idx int, acc int, tree []int) (int, int) {
	nNode := tree[idx]
	nMeta := tree[idx+1]
	idx += 2

	for n := 0; n < nNode; n++ {
		idx, acc = rec(idx, acc, tree)
	}

	for n := 0; n < nMeta; n++ {
		acc += tree[idx+n]
	}

	return idx + nMeta, acc
}

func main() {
	input, err := ioutil.ReadAll(os.Stdin)
	if err != nil {
		panic(err)
	}
	start := time.Now()
	res := run(string(input))
	fmt.Printf("_duration:%f\n", time.Now().Sub(start).Seconds()*1000)
	fmt.Println(res)
}

File: day-08/part-1/badouralix.py
from runners.python import SubmissionPy


class BadouralixSubmission(SubmissionPy):

    def run(self, s):
        split = list(map(int, s.split(" ")))
        _, acc = self.rec(split, 0)
        return acc

    def rec(self, ll, acc):
        nodes = ll[0]
        metadata = ll[1]
        ll = ll[2:]
        for n in range(nodes):
            ll, acc = self.rec(ll, acc)
        for i in range(metadata):
            acc += ll[i]
        return ll[metadata:], acc
